metrics/mean_iou/README.md

```diff
@@ -27,13 +27,13 @@ For binary (two classes) or multi-class segmentation, the *mean IoU* of the imag
 
 ## How to Use
 
-The Mean IoU metric takes two numeric arrays as input corresponding to the predicted and ground truth segmentations:
+The Mean IoU metric takes two lists of numeric 2D arrays as input corresponding to the predicted and ground truth segmentations:
 ```python
 >>> import numpy as np
 >>> mean_iou = evaluate.load("mean_iou")
 >>> predicted = np.array([[2, 2, 3], [8, 2, 4], [3, 255, 2]])
 >>> ground_truth = np.array([[1, 2, 2], [8, 2, 1], [3, 255, 1]])
->>> results = mean_iou.compute(predictions=predicted, references=ground_truth, num_labels=10, ignore_index=255)
+>>> results = mean_iou.compute(predictions=[predicted], references=[ground_truth], num_labels=10, ignore_index=255)
```

Comment:
Would this be a breaking change? Cause currently the metric works when passing 2D arrays for predicted and ground_truth, but this PR would require it to be a list of 2D arrays?

Reply:
It seems like it doesn't work with 2D arrays originally, only with a list of them.

I tested it locally and got the following error:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/piakubovskii/Projects/evaluate/.venv/lib/python3.10/site-packages/evaluate/module.py", line 450, in compute
    self.add_batch(**inputs)
  File "/Users/piakubovskii/Projects/evaluate/.venv/lib/python3.10/site-packages/evaluate/module.py", line 541, in add_batch
    raise ValueError(error_msg) from None
ValueError: Predictions and/or references don't match the expected format.
Expected format: {'predictions': Sequence(feature=Sequence(feature=Value(dtype='uint16', id=None), length=-1, id=None), length=-1, id=None), 'references': Sequence(feature=Sequence(feature=Value(dtype='uint16', id=None), length=-1, id=None), length=-1, id=None)},
Input predictions: [[  2   2   3]
 [  8   2   4]
 [  3 255   2]],
Input references: [[  1   2   2]
 [  8   2   1]
 [  3 255   1]]
 ```
 
 There is also an issue opened 3 weeks ago with the same error:
 https://github.com/huggingface/evaluate/issues/563


Maybe backward compatibility have been broken earlier